spec/models/shared/availability_status.rb

```diff
@@ -0,0 +1,80 @@
+RSpec.shared_examples "availability_status" do
+  # record variable comes from caller
+  describe "#before_update" do
+    let(:timestamp)                 { Time.current }
+    let(:new_timestamp)             { Time.current + 1.hour }
+    let(:available_status)          { "available" }
+    let(:unavailable_status)        { "unavailable" }
+    let(:new_name)                  { "new_name" }
+    let(:availability_status_error) { "availability_status_error" }
+
+    let(:ignored_attributes) do
+      new_attributes = {
+        :availability_status => unavailable_status,
+        :last_checked_at     => new_timestamp,
+        :last_available_at   => new_timestamp
+      }
+
+      if record.respond_to?(:updated_at)
+        new_attributes[:updated_at] = new_timestamp
+      end
+
+      if record.respond_to?(:availability_status_error)
+        new_attributes[:availability_status_error] = availability_status_error
+      end
+
+      if record.respond_to?(:name)
+        new_attributes[:name] = new_name
+      end
+
+      new_attributes
+    end
+
+    before do
+      record.availability_status = available_status
+      record.last_checked_at = timestamp
+    end
+
+    context "with changes" do
+      it "sets availability_status to nil" do
+        record.update!({:id=>record.id + 1})
```

Comment:
Let's change this to `:name` just to be more realistic. We would never actually update the ID.

Reply:
@lindgrenj6 I was talking with Martin before lockdown and we agreed that the `name` attribute change should not clear the availability_status and we putted him into the [ignore list](https://github.com/RedHatInsights/sources-api/blob/c968069fe5f6f3bf2278d1c8b6ba378dbb418b49/app/models/concerns/availability_status_concern.rb#L10-L17). Also I was checking the database and the only attributes that comes out from the union of all related models(Endpoints, Applications, Sources and Authentications) are `id` and `tenant_id`. Since this is shared example group, I think we should either keep with one of them or make specific change for every model similar to the record forwarding [here](https://github.com/RedHatInsights/sources-api/blob/c968069fe5f6f3bf2278d1c8b6ba378dbb418b49/spec/models/authentication_spec.rb#L5). What do you think Jacob?

Reply:
@pkomanek well it doesn't have to be name, that makes sense. lets just update something in the test that isn't `:id` since updating id will never happen in real life. `availability_status` would work.